storagenode/peer.go

```diff
@@ -107,22 +113,29 @@ func New(log *zap.Logger, full *identity.FullIdentity, db DB, config Config) (*P
 			Id:   peer.ID(),
 			Type: pb.NodeType_STORAGE,
 			Address: &pb.NodeAddress{
-				Address: config.ExternalAddress,
+				Transport: pb.NodeTransport_TCP_TLS_GRPC,
+				Address:   config.ExternalAddress,
 			},
 			Metadata: &pb.NodeMetadata{
 				Email:  config.Operator.Email,
 				Wallet: config.Operator.Wallet,
 			},
 		}
 
+		// TODO(coyle): I'm thinking we just remove this function and grab from the config.
+		in, err := kademlia.GetIntroNode(config.BootstrapAddr)
+		if err != nil {
+			return nil, errs.Combine(err, peer.Close())
+		}
+
 		kdb, ndb := peer.DB.RoutingTable()
 		peer.RoutingTable, err = kademlia.NewRoutingTable(peer.Log.Named("routing"), self, kdb, ndb)
 		if err != nil {
 			return nil, errs.Combine(err, peer.Close())
 		}
 
 		// TODO: reduce number of arguments
-		peer.Kademlia, err = kademlia.NewWith(peer.Log.Named("kademlia"), self, nil, peer.Identity, config.Alpha, peer.RoutingTable)
+		peer.Kademlia, err = kademlia.NewWith(peer.Log.Named("kademlia"), self, []pb.Node{*in}, peer.Identity, config.Alpha, peer.RoutingTable)
```

Comment:
why do we want a node now ?

Reply:
Because initially it was only used in testplanet which sets the bootstrap nodes via `SetBootstrapNodes`, because at the creation time, we don't have the necessary info yet.

However storagenode binary doesn't specify it separately. I've considered always specify it outside, but it probably makes more sense to pass it in this way.